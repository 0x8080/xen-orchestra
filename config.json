// Vendor config: DO NOT TOUCH!
//
// See sample.config.yaml to override.
{
  "http": {
    "listen": [
      {
        "port": 80
      }
    ],
    "mounts": {}
  },
  "datadir": "/var/lib/xo-server/data",

  // Should users be created on first sign in?
  //
  // Necessary for external authentication providers.
  "createUserOnFirstSignin": true,

  // Whether API logs should contains the full request/response on
  // errors.
  //
  // This is disabled by default for performance (lots of data) and
  // security concerns (avoiding sensitive data in the logs) but can
  // be turned for investigation by the administrator.
  "verboseApiLogsOnErrors": false
}
