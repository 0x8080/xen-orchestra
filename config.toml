iso = 'Windows7Ultimate.iso'
network = 'Pool-wide network associated with eth0'
pool = 'lab3'
pvVm = 'xo-test-pv'
vmToMigrate = 'souad'
xoServerUrl = 'localhost:9000'
vmIdXoTest = '0342407f-9727-4f12-83c2-9d0c458916c2'

# powerOnMode has to be not empty
host1 = 'lab1'

[adminCredentials]
  email = 'admin@admin.net'
  password = 'admin'

[templates]
debian = 'Debian Wheezy 7.0 (64-bit)'
otherConfig = 'Other install media'
centOS = 'CentOS 7'

[slaveServer]
host = '192.168.100.1'
username = 'root'
password = 'qwerty'
autoConnect = false

[[xenServers]]
host = '192.168.100.3'
username = 'root'
password = 'qwerty'

[[xenServers]]
host = '192.168.100.2'
username = 'root'
password = 'qwerty'
