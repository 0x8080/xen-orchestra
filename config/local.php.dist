<?php
/**
 * This file is a part of Xen Orchestra Server.
 *
 * Xen Orchestra Server is free software: you can redistribute it
 * and/or modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Xen Orchestra Server is distributed in the hope that it will be
 * useful, but WITHOUT ANY WARRANTY; without even the implied warranty
 * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Xen Orchestra Server. If not, see
 * <http://www.gnu.org/licenses/>.
 *
 * @author Julien Fontanet <julien.fontanet@vates.fr>
 * @license http://www.gnu.org/licenses/gpl-3.0-standalone.html GPLv3
 *
 * @package Xen Orchestra Server
 *
 * Local Variables:
 * mode: php
 * End:
 */

/* Local settings
 *
 * This file contains all settings related to this current
 * installation of Xen Orchestra Server.
 *
 * You MUST define the following settings for which no default
 * values exists.
 *
 * But, you MAY override any settings which already exists in
 * “global.php”.
 */
return array(

	// For now, XCP servers/pool masters must be defined in this file.
	'xcp' => array(
		//'pool 1' => array(
		//	'url'      => 'https://xcp1.example.net',
		//	'username' => 'username',
		//	'password' => 'password'
		//),
	),
);


