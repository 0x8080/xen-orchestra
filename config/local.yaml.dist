# Note: Relative paths will be resolved from XO-Server's directory.

# Configuration of the embedded HTTP server.
http:

  # Hosts & ports on which to listen.
  listen:
    # Basic HTTP.
    -
      # Address on which the server is listening on.
      #
      # Sets it to '127.0.0.1' to listen only on the local host.
      #
      # Default: '0.0.0.0' (all addresses)
      #host: '127.0.0.1'

      # Port on which the server is listening on.
      #
      # Default: undefined
      port: 80

      # Instead of `host` and `port` a path to a UNIX socket may be
      # specified (overrides `host` and `port`).
      #
      # Default: undefined
      #socket: './http.sock'

    # Basic HTTPS.
    # -
    #   # The only difference is the presence of the certificate and the
    #   # key.

    #   #host: '127.0.0.1'
    #   port: 443

    #   # File containing the certificate (PEM format).
    #   #
    #   # Default: undefined
    #   certificate: './certificate.pem'

    #   # File containing the private key (PEM format).
    #   #
    #   # If the key is encrypted, the passphrase will be asked at
    #   # server startup.
    #   #
    #   # Default: undefined
    #   key: './key.pem'

  # List of files/directories which will be served.
  mounts:
    '/': './public/http/'

# Connection to the Redis server.
redis:
    # Syntax: tcp://[db[:password]@]hostname[:port]
    #
    # Default: tcp://localhost:6379
    #uri: ''
