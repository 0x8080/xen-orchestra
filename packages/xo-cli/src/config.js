'use strict'

// ===================================================================

const promisify = require('bluebird').promisify

const readFile = promisify(require('fs').readFile)
const writeFile = promisify(require('fs').writeFile)

const get = require('lodash/get')
const mkdirp = require('mkdirp')
const unset = require('lodash/unset')
const xdgBasedir = require('xdg-basedir')

// ===================================================================

const configPath = xdgBasedir.config + '/xo-cli'
const configFile = configPath + '/config.json'

// ===================================================================

const load = (exports.load = function () {
  return readFile(configFile)
    .then(JSON.parse)
    .catch(function () {
      return {}
    })
})

exports.get = function (path) {
  return load().then(function (config) {
    return get(config, path)
  })
}

const save = (exports.save = function (config) {
  return mkdirp(configPath).then(function () {
    return writeFile(configFile, JSON.stringify(config))
  })
})

exports.set = function (data) {
  return load().then(function (config) {
    return save(Object.assign(config, data))
  })
}

exports.unset = function (paths) {
  return load().then(function (config) {
    ;[].concat(paths).forEach(function (path) {
      unset(config, path)
    })
    return save(config)
  })
}
