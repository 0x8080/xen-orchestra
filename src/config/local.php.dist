<?php
/**
 * This file is a part of Xen Orchestra Web.
 *
 * Xen Orchestra Web is free software: you can redistribute it
 * and/or modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Xen Orchestra Web is distributed in the hope that it will be
 * useful, but WITHOUT ANY WARRANTY; without even the implied warranty
 * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Xen Orchestra Web. If not, see
 * <http://www.gnu.org/licenses/>.
 *
 * @author Julien Fontanet <julien.fontanet@vates.fr>
 * @license http://www.gnu.org/licenses/gpl-3.0-standalone.html GPLv3
 *
 * @package Xen Orchestra Web
 *
 * Local Variables:
 * mode: php
 * End:
 */

/* Local settings
 *
 * This file contains all settings related to this current
 * installation of Xen Orchestra Web.
 *
 * You MUST define the following settings for which no default
 * values exists.
 *
 * But, you MAY override any settings which already exists in
 * “global.php”.
 */
return array(
);
