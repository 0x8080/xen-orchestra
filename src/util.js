import {Xo} from 'xo-lib'
import {find, forEach, map, cloneDeep} from 'lodash'
import expect from 'must'

export async function getConfig () {
  return {
    adminCredentials: {
      email: 'admin@admin.net',
      password: 'admin'
    },
    xoServerUrl: 'localhost:9000',
    xenServer1: {
      host: '192.168.1.3',
      username: 'root',
      password: 'qwerty'
    }
  }
}

export async function getConnection ({
  credentials
} = {}) {
  const config = await getConfig()
  const xo = new Xo(config.xoServerUrl)
  await xo.signIn(
    credentials === undefined ?
      config.adminCredentials :
      credentials
  )

  // Injects waitObject()
  //
  // TODO: integrate in xo-lib.
  const watchers = {}
  xo.waitObject = (id) => {
    return new Promise(resolve => {
      watchers[id] = resolve
    })
  }

  const onUpdate = objects => {
    forEach(objects, (object, id) => {
      if (watchers[id]) {
        watchers[id](object)
        delete watchers[id]
      }
    })
  }
  xo.objects.on('add', onUpdate)
  xo.objects.on('update', onUpdate)
  xo.objects.on('remove', onUpdate)

  xo.getOrWaitObject = async function (id) {
    const object = this.objects.all[id]
    if (object) {
      return object
    }

    return await this.waitObject(id)
  }

  return xo
}

// =================================================================

export async function getAllUsers (xo) {
  return await xo.call('user.getAll')
}

export async function getUser (xo, id) {
  const users = await getAllUsers(xo)
  return find(users, {id: id})
}

export async function createUser (xo, userIds, params) {
  const userId = await xo.call('user.create', params)
  userIds.push(userId)
  return userId
}

export async function deleteUsers (xo, userIds) {
  await Promise.all(map(
    userIds,
    userId => xo.call('user.delete', {id: userId})
  ))
}

// ==================================================================

export function getAllHosts (xo) {
  return xo.objects.indexes.type.host
}

export function getOneHost (xo) {
  const hosts = getAllHosts(xo)
  for (const id in hosts) {
    return hosts[id]
  }

  throw new Error('no hosts found')
}

// ==================================================================

export function deepDelete (obj, path) {
  const lastIndex = path.length - 1
  for (let i = 0; i < lastIndex; i++) {
    obj = obj[path[i]]

    if (typeof obj !== 'object' || obj === null) {
      return
    }
  }
  delete obj[path[lastIndex]]
}

export function almostEqual (actual, expected, ignoredAttributes) {
  actual = cloneDeep(actual)
  expected = cloneDeep(expected)
  forEach(ignoredAttributes, (ignoredAttribute) => {
    deepDelete(actual, ignoredAttribute.split('.'))
    deepDelete(expected, ignoredAttribute.split('.'))
  })
  expect(actual).to.be.eql(expected)
}

export async function waitObjectState (xo, id, predicate) {
  let obj = xo.objects.all[id]
  while (true) {
    try {
      await predicate(obj)
      return
    } catch (_) {}
    // If failed, wait for next object state/update and retry.
    obj = await xo.waitObject(id)
  }
}
