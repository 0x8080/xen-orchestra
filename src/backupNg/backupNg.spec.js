/* eslint-env jest */

import { omit } from "lodash";

import config from "../_config";
import randomId from "../_randomId";
import { xo } from "../util";

describe("backupNg", () => {
  let defaultBackupNg;

  beforeAll(() => {
    defaultBackupNg = {
      name: "default-backupNg",
      mode: "full",
      vms: {
        id: config.vmIdXoTest,
      },
      settings: {
        "": {
          reportWhen: "always",
        },
      },
    };
  });

  describe(".create() :", () => {
    it("creates a new backup job without schedules", async () => {
      const backupNg = await xo.createTempBackupNgJob(defaultBackupNg);
      expect(omit(backupNg, "id", "userId")).toMatchSnapshot();
      expect(backupNg.userId).toBe(xo._user.id);
    });

    it("creates a new backup job with schedules", async () => {
      const scheduleTempId = randomId();
      const { id: jobId } = await xo.createTempBackupNgJob({
        ...defaultBackupNg,
        schedules: {
          [scheduleTempId]: {
            name: "scheduleTest",
            cron: "0 * * * * *",
          },
        },
        settings: {
          ...defaultBackupNg.settings,
          [scheduleTempId]: { snapshotRetention: 1 },
        },
      });

      const backupNgJob = await xo.call("backupNg.getJob", { id: jobId });

      expect(omit(backupNgJob, "id", "userId", "settings")).toMatchSnapshot();
      expect(backupNgJob.userId).toBe(xo._user.id);

      const settingKeys = Object.keys(backupNgJob.settings);
      expect(settingKeys.length).toBe(2);
      const scheduleId = settingKeys.find(key => key !== "");
      expect(backupNgJob.settings[scheduleId]).toEqual({
        snapshotRetention: 1,
      });

      const schedule = await xo.call("schedule.get", { id: scheduleId });
      expect(omit(schedule, "id", "jobId")).toMatchSnapshot();
      expect(schedule.jobId).toBe(jobId);
    });
  });
});
