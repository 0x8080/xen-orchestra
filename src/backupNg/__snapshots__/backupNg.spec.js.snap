// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`backupNg .create() : creates a new backup job with schedules 1`] = `
Object {
  "mode": "full",
  "name": "default-backupNg",
  "type": "backup",
  "vms": Object {
    "id": "0342407f-9727-4f12-83c2-9d0c458916c2",
  },
}
`;

exports[`backupNg .create() : creates a new backup job with schedules 2`] = `
Object {
  "cron": "0 * * * * *",
  "enabled": false,
  "name": "scheduleTest",
}
`;

exports[`backupNg .create() : creates a new backup job without schedules 1`] = `
Object {
  "mode": "full",
  "name": "default-backupNg",
  "settings": Object {
    "": Object {
      "reportWhen": "always",
    },
  },
  "type": "backup",
  "vms": Object {
    "id": "0342407f-9727-4f12-83c2-9d0c458916c2",
  },
}
`;

exports[`backupNg .runJob() : fails trying to run a backup job without schedule 1`] = `[JsonRpcError: invalid parameters]`;
